app/controllers/BaseNavigationController.scala

```diff
@@ -59,10 +63,24 @@ trait BaseNavigationController extends BaseController with Logging {
       } yield updatedAnswers
     }
 
-  private def save[A](page: QuestionPage[A], answer: A)
-                     (implicit request: DataRequest[_], format: Format[A]): Future[UserAnswers] =
-    save(page, answer, request.userAnswers)
-
+  private[controllers] def markErrorAsFixedIfPresent(page: QuestionPage[_])(implicit request: DataRequest[_]): UserAnswers = {
+    if(request.userAnswers.haveAllSubmissionErrorsBeenFixed) {
+     request.userAnswers
+    } else {
+      val indexOfError = Try(page match {
+        case LocalReferenceNumberPage(_) => request.userAnswers.submissionFailures.indexWhere(_.errorType == localReferenceNumberError)
```

Comment:
Just checking I understand this - for future pages, would you just add extra cases to this function?

Reply:
Yeah, exactly that for 1-1 mappings between page and error. Where a page could have multiple errors (if it's possible?) then perhaps this needs to be enhanced